### PR TITLE
Refactor the `TestCredsFromEnv` function into internal package

### PR DESCRIPTION
**Pull request by thunderboltsid**
This is done so the funciton can be shared across all API packages (karbon, v3, etc.). The function also now reads a prism developer config from user home dir (~/.prism-dev.yaml) in order to get developer specific prism environment configuration. Alternate file path for prism developer config can be specified using `PRISM_DEV_CONFIG` environment variable. The prism developer config file looks like this:
```
prismCentral:
  endpoint: <endpoint_for_prism_central>
  port: <port_for_prism_central>
  username: <username_for_prism_central>
  password: <password_for_prism_central>
  insecure: <true_or_false>
```

The developer can populate the file with whichever prism enviornment they want to use for keploy recording of the API calls for use as data mocks in unit tests.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Developing on the project
+
+## Building the project locally
+- run `git clone repo_uri.git` to clone the project to your machine
+- run `make build` to build the project locally.
+- run `make generate` to generate the deepcopy methods for types with kubebuilder annotations.
```
Comment (deepakm-ntnx) [inline DEVELOPMENT.md:6]: why is this required?

File: internal/testhelpers/creds.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	nutanixPortDummyValue     = "9440"
+	nutanixUsernameDummyValue = "test"
+	nutanixPasswordDummyValue = "test"
+	nutanixInsecureDummyValue = false
+	prismDevConfigFileName    = ".prism-dev.yaml"
+	prismDevConfigFileEnvVar  = "PRISM_DEV_CONFIG"
+)
+
+type prismCentral struct {
+	Endpoint              string `yaml:"endpoint"`
```
Comment (deepakm-ntnx) [inline internal/testhelpers/creds.go:28]: can we remove yaml tags as it wont be used here

Comment (thunderboltsid) [reply internal/testhelpers/creds.go:28]: so, when we do a `yaml.Unmarshal` how will the unmarshaller know which field to unmarshal where?

**Review by deepakm-ntnx — APPROVED** [review]

**Review by tuxtof — COMMENTED** [review]
LGTM

**Review by tuxtof — APPROVED** [review]
